Remove empty statement

clement26695 opened the pull request.

bd82 commented on packages/prettier-plugin-java/test/comments/comments-blocks-and-statements/_output.java:
```diff
@@ -3,7 +3,7 @@
   public void myFunction(int arg1) {
     try {
-      ;// Empty Statement
+      // Empty Statement
```
I am with @Shaolans here. I think this represents several unique edge cases.
I would like to see most of those tested, perhaps even with a separate test input/output pair.

edge cases:
- empty statement with comment.
- empty statement in for/do header.
- empty statement without comment

clement26695 commented:
I added some tests and refactored the code to not modify the CST during processing.

There is still an issue in the tests I need to solve (issue on stable reformating)

bd82 commented on packages/prettier-plugin-java/src/printers/printer-utils.js:
```diff
@@ -156,8 +156,44 @@ function findDeepElementInPartsArray(item, elt) {
+}
+
+function isExplicitLambdaParameter(ctx) {
+  return (
```
JavaScript needs Kotlin's null safety features 😄 
https://kotlinlang.org/docs/reference/null-safety.html#safe-calls

bd82 commented on packages/prettier-plugin-java/test/empty_statement/_input.java:
```diff
@@ -1,5 +1,27 @@
+
+    for (;;)/*test*/;
+
+    for (;;);/*test*/
```
worth checking edge case of comments _between_ the for loop's semi colons?

clement26695 replied:
These semi colons are not considered as empty statements, so that's fine, IMO. But I can add some, that can not harm !
